blog-content/the-complete-guide-to-opentelemetry-in-next-js.md

```diff
@@ -0,0 +1,394 @@
+---
+title: The complete guide to OpenTelemetry in Next.js
+createdAt: 2025-02-10T12:00:00.000Z
+readingTime: 21
+authorFirstName: Vadim
+authorLastName: Korolik
+authorTitle: CTO @ Highlight
+authorTwitter: 'https://twitter.com/vkorolik'
+authorLinkedIn: 'https://www.linkedin.com/in/vkorolik/'
+authorGithub: 'https://github.com/Vadman97'
+authorWebsite: 'https://vadweb.us'
+authorPFP: 'https://lh3.googleusercontent.com/a-/AOh14Gh1k7XsVMGxHMLJZ7qesyddqn1y4EKjfbodEYiY=s96-c'
+tags: 'Engineering, Backend, Observability'
+metaTitle: The complete guide to OpenTelemetry in Next.js
+---
+
+```hint
+Highlight.io is an [open source](https://github.com/highlight/highlight) monitoring platform. If you’re interested in learning more, get started at [highlight.io](https://highlight.io).
+```
+<br/>
+
+OpenTelemetry is an important specification that defines how we send telemetry data to observability backends like Highlight.io, Grafana, and others. OpenTelemetry is great because it is vendor agnostic, and can be used with several observability backends. If you're new to OpenTelemetry, you can learn more about it [here](https://www.youtube.com/watch?v=ASgosEzG4Pw). 
+
+
+Today, we'll go through a complete guide to using OpenTelemetry in Python, including the high-level concepts as well as how to send traces and logs to your OpenTelemetry backend of choice.
+
+
+## Setting Up OpenTelemetry for Next.js: Tracing, Logging, and Metrics
+
+Observability is crucial for understanding the performance and behavior of your Next.js application. OpenTelemetry (OTel) provides a robust framework for tracing, logging, and metrics collection, allowing you to gain deep insights into how your application operates.
+
+In this guide, we’ll walk through setting up OpenTelemetry in a Next.js project, covering:
+- Tracing: Capturing distributed traces for API requests and page transitions
+- Logging: Collecting structured logs that correlate with traces
+- Metrics: Exporting performance and custom application metrics
+- Built-in Spans: Leveraging Next.js’s automatic spans and debugging with NEXT_OTEL_VERBOSE
+- Exception Tracking: Capturing errors within traces
+- Simplifying Setup with @vercel/otel
+
+By the end of this tutorial, you’ll have OpenTelemetry integrated into your Next.js project with traces, logs, and metrics exported to your preferred backend.
```

Comment:
Nit, but feels duplicative to the previous sentence. Maybe we could say something about how you'll have all the observability data you need to be proactively notified if something goes wrong, troubleshoot issues quickly, find and fix performance bottlenecks, etc.?